python/libraries/websocket.py

```diff
@@ -123,7 +123,10 @@ async def on_message():
                             close = await close.eval()
                         if close:
                             await websocket.close()
-                            on_close.set_result(None) # Mark `on_close` as done
+                            try:
+                                on_close.set_result(None) # Mark `on_close` as done
+                            except:
+                                pass
```

Comment:
A better solution here would be to [check if `on_close` is done](https://docs.python.org/3/library/asyncio-future.html#asyncio.Future.done)

```suggestion
                            if not on_close.done():
                                on_close.set_result(None) # Mark `on_close` as done
```